packages/atomic/src/components/commerce/result-template-components/atomic-product-link/atomic-product-link.tsx

```diff
@@ -0,0 +1,105 @@
+import {isUndefined} from '@coveo/bueno';
+import {InteractiveProduct} from '@coveo/headless/commerce';
+import {Product} from '@coveo/headless/commerce';
+import {Component, h, Prop, Element} from '@stencil/core';
+import {buildCustomEvent} from '../../../../utils/event-utils';
+import {
+  InitializableComponent,
+  InitializeBindings,
+} from '../../../../utils/initialization-utils';
+import {getDefaultSlotFromHost} from '../../../../utils/slot-utils';
+import {getAttributesFromLinkSlot} from '../../../common/item-link/attributes-slot';
+import {LinkWithItemAnalytics} from '../../../common/item-link/item-link';
+import {Bindings} from '../../../search/atomic-search-interface/atomic-search-interface';
+import {
+  InteractiveProductContext,
+  ProductContext,
+} from '../product-template-decorators';
+
+/**
+ * The `atomic-product-link` component automatically transforms a search product title into a clickable link that points to the original item.
```

Comment:
Mark this one as `@internal`, and rebuild to regen angular/react output generated code.

For now, we want all commerce components marked as such.